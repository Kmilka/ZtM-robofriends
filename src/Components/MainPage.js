import React, { Component } from "react";
import CardList from "../Components/CardList";
import SearchBox from "../Components/SearchBox.js";
import Scroll from "../Components/Scroll.js";
import Header from "../Components/Header";
import "./MainPage.css";
import CounterButton from "../Components/CounterButton";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    const filteredRobots = robots.filter((robot) => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
    return filteredRobots;
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1>Loading</h1>
    ) : (
      <div className="tc">
        <Header />
        <CounterButton color={"red"} />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <CardList robots={this.filterRobots()} />
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
