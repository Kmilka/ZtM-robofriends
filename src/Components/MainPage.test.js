import { shallow } from "enzyme";
import React from "react";
import MainPage from "../Components/MainPage.js";

describe("MainPage tests", () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      searchField: "",
      isPending: false,
      robots: [],
      error: "",
    };
    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("expect to render MainPage", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("filters robots correctly", () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
    const mockPropsToFilter = {
      onRequestRobots: jest.fn(),
      searchField: "a",
      isPending: false,
      robots: [
        {
          id: 3,
          name: "john",
          email: "mymail@com",
        },
        {
          id: 4,
          name: "melody",
          email: "mamamia@melody.net",
        },
        {
          id: 1,
          name: "ana",
          email: "ananas@tata.ru",
        },
      ],
    };
    expect(
      shallow(<MainPage {...mockPropsToFilter} />)
        .instance()
        .filterRobots()
    ).toEqual([
      {
        id: 1,
        name: "ana",
        email: "ananas@tata.ru",
      },
    ]);

    const mockPropsNotToFilter = {
      onRequestRobots: jest.fn(),
      searchField: "dummy",
      isPending: false,
      robots: [
        {
          id: 3,
          name: "john",
          email: "mymail@com",
        },
        {
          id: 4,
          name: "melody",
          email: "mamamia@melody.net",
        },
        {
          id: 1,
          name: "ana",
          email: "ananas@tata.ru",
        },
      ],
    };
    expect(
      shallow(<MainPage {...mockPropsNotToFilter} />)
        .instance()
        .filterRobots()
    ).toEqual([]);
  });

  it("renders correct pending-dependent component", () => {
    const mockPropsPendingTrue = {
      onRequestRobots: jest.fn(),
      searchField: "",
      isPending: true,
      robots: [],
      error: "",
    };
    expect(
      shallow(<MainPage {...mockPropsPendingTrue} />).contains(<h1>Loading</h1>)
    ).toEqual(true);
  });
});
