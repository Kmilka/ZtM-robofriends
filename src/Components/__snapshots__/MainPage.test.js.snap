// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage tests expect to render MainPage 1`] = `ShallowWrapper {}`;
